projects/core/src/cart/event/cart-event.collector.ts

```diff
@@ -0,0 +1,54 @@
+import { Injectable } from '@angular/core';
+import { map } from 'rxjs/operators';
+import { EventService } from '../../event/event.service';
+import { AbstractTmsEventCollector } from '../../event/tms.event';
+import {
+  CartAddEntryEvent,
+  CartAddEntryFailEvent,
+  CartAddEntrySuccessEvent,
+  CartRemoveEntrySuccessEvent,
+  CartUpdateEntrySuccessEvent,
+} from './cart.events';
+
+/**
+ * Registers cart event collectors
+ */
+@Injectable({ providedIn: 'root' })
+export class CartEventCollector extends AbstractTmsEventCollector {
+  protected cartAddEntryEvent$ = this.eventsService
+    .get(CartAddEntryEvent)
+    .pipe(map((event) => this.mapEvent(CartAddEntryEvent.type, event)));
+
+  protected cartAddEntrySuccessEvent$ = this.eventsService
+    .get(CartAddEntrySuccessEvent)
+    .pipe(map((event) => this.mapEvent(CartAddEntrySuccessEvent.type, event)));
+
+  protected cartAddEntryFailEvent$ = this.eventsService
+    .get(CartAddEntryFailEvent)
+    .pipe(map((event) => this.mapEvent(CartAddEntryFailEvent.type, event)));
+
+  protected cartRemoveEntrySuccessEvent$ = this.eventsService
+    .get(CartRemoveEntrySuccessEvent)
+    .pipe(
+      map((event) => this.mapEvent(CartRemoveEntrySuccessEvent.type, event))
+    );
+
+  protected cartUpdateEntrySuccessEvent$ = this.eventsService
+    .get(CartUpdateEntrySuccessEvent)
+    .pipe(
+      map((event) => this.mapEvent(CartUpdateEntrySuccessEvent.type, event))
+    );
+
+  protected events$ = [
+    this.cartAddEntryEvent$,
+    this.cartAddEntrySuccessEvent$,
+    this.cartAddEntryFailEvent$,
+    this.cartRemoveEntrySuccessEvent$,
+    this.cartUpdateEntrySuccessEvent$,
+  ];
+
+  constructor(protected eventsService: EventService) {
+    super(eventsService);
+    this.registerEvent(...this.events$);
```

Comment:
Will it work if moved to superclass?

Reply:
Sadly, it won't.
I've tried it already, and this is what I got: 
<img width="721" alt="Screen Shot 2021-01-11 at 09 55 29" src="https://user-images.githubusercontent.com/11561723/104161457-310c9a80-53f3-11eb-9c0c-df072fea319b.png">